libbeat/esleg/eslegclient/bulkapi.go

```diff
@@ -99,6 +101,8 @@ func (conn *Connection) SendMonitoringBulk(
 		return nil, nil
 	}
 
+	mergedParams := mergeParams(conn.ConnectionSettings.Parameters, params)
```

Comment:
Looks like we have the same logic several times, perhaps worth cleaning up in the future ;-)

Reply:
I wanted to make it close to the HTTP, but this is not currently possible with the way the HTTP client is composed today we lose context where the we do the actual URL generation. :/